### ISSUE TITLE
Files with conditionals break if they evaluate to nothing.

### ISSUE THREAD

**Issue by pydanny** [audreyr/cookiecutter#267]
I discovered this when working on cookiecutter-pymodule. If I create a folder with a conditional:
```
mkdir '{% if cookiecutter.test_on_windows != "no" %}appveyor{% endif %}
```
* If `True`: cookiecutter generates the folder with the files inside
* If `False`: cookiecutter doesn't create the folder.

However, if I create a file with a conditional:

```
touch '{% if cookiecutter.test_on_windows != "no" %}appveyor.yml{% endif %}'
```

On generation it triggers a python IO read exception:

```
cookiecutter ../cookiecutter-pymodule/
module_name (default is "The name of the to-be-packaged model.")? papa5
full_name (default is "Daniel Greenfeld")? 
email (default is "pydanny@gmail.com")? 
github_username (default is "pydanny")? 
short_description (default is "This package does X")? 
release_date (default is "2014-01-11")? 
console_script_name (default is "")? 
test_on_windows (default is "no")? 
Traceback (most recent call last):
  File "/Users/danny/.envs/pelican/bin/cookiecutter", line 9, in <module>
    load_entry_point('cookiecutter==0.7.2', 'console_scripts', 'cookiecutter')()
  File "/Users/danny/.envs/pelican/lib/python2.7/site-packages/cookiecutter/main.py", line 130, in main
    cookiecutter(args.input_dir, args.checkout, args.no_input)
  File "/Users/danny/.envs/pelican/lib/python2.7/site-packages/cookiecutter/main.py", line 70, in cookiecutter
    context=context
  File "/Users/danny/.envs/pelican/lib/python2.7/site-packages/cookiecutter/generate.py", line 196, in generate_files
    generate_file(project_dir, infile, context, env)
  File "/Users/danny/.envs/pelican/lib/python2.7/site-packages/cookiecutter/generate.py", line 117, in generate_file
    with io.open(outfile, 'w', encoding="utf-8") as fh:
IOError: [Errno 21] Is a directory: '/Users/danny/pydanny/cookiecutter-pymodule/papa5/./'
```

This is happening because internally Python is trying to do:

```python
>>> open('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IOError: [Errno 2] No such file or directory: ''
```

**Comment by iljabauer** [comment]
I submitted a pull request with a fix for this (1717f1d513236f07536d6042771bf44277e0bc06)

**Comment by audreyr** [comment]
Closed by #444.
